Automattic/jetpack
# JITMs: add redux structure and REST API client method to be used by the JITM component of the Jetpack Dashboard (Spin-off of #10759).

This PR is an extract of #10759 in order to simplify that PR and scope it to UI changes.

Author: @jeherve 

#### Changes proposed in this Pull Request:

* Introduces `_inc/client/state/jitm` and files for reducer, actions and tests.
* Introduces method `fetchJitm` in the REST API client.

#### Testing instructions:


* Checkout this branch
* Run `yarn test-client _inc/client/state/jitm/test/*`
* Confirm tests pass

#### Proposed changelog entry for your changes:

None needed
